community/kernel/src/main/java/org/neo4j/kernel/impl/transaction/command/PhysicalLogNeoCommandReaderV2_1.java

```diff
@@ -309,7 +310,7 @@ public boolean visitRelationshipGroupCommand( Command.RelationshipGroupCommand c
             {
                 throw new IOException( "Illegal in use flag: " + inUseByte );
             }
-            int type = channel.getShort();
+            int type = shortToUnsignedInt( channel.getShort() );
```

Comment:
That is true for `V2_1` reader. But it is possible to start up 2.2.10 on 2.2.4 non-cleanly shutdown store. This is not an upgrade and we need to read 2.2.4 log with `V2_2_4` reader while our own reader is `V2_2_10`. That is my understanding.